src/frontend/src/modals/flowLogsModal/index.tsx

```diff
@@ -13,22 +14,53 @@ export default function FlowLogsModal({
 }: FlowSettingsPropsType): JSX.Element {
   const currentFlowId = useFlowsManagerStore((state) => state.currentFlowId);
 
+  const [pageIndex, setPageIndex] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
   const [columns, setColumns] = useState<Array<ColDef | ColGroupDef>>([]);
   const [rows, setRows] = useState<any>([]);
 
-  const { data, isLoading, refetch } = useGetTransactionsQuery({
+  const {
+    data: TransactionData,
+    isLoading,
+    refetch,
+  } = useGetTransactionsQuery({
     id: currentFlowId,
+    params: {
+      page: pageIndex,
+      size: pageSize,
+    },
     mode: "union",
   });
 
+  const data = {
+    rows: TransactionData?.rows ?? [],
+    columns: TransactionData?.columns ?? [],
+    pagination: {
+      page: TransactionData?.pagination?.page ?? 1,
+      size: TransactionData?.pagination?.size ?? 10,
+      total: TransactionData?.pagination?.total ?? 0,
+      pages: TransactionData?.pagination?.pages ?? 0,
+    },
+  };
+
   useEffect(() => {
-    if (data) {
+    if (TransactionData) {
       const { columns, rows } = data;
       setColumns(columns.map((col) => ({ ...col, editable: true })));
       setRows(rows);
     }
-    if (open) refetch();
-  }, [data, open, isLoading]);
+  }, [TransactionData]);
```

Comment:
Shouldn't the variables named `TransactionData` be `data` instead?

Reply:
@ogabrielluiz  Thank you for pointing that out.I have removed the redundant `TransactionData` variable and updated the code accordingly.